### ISSUE TITLE
Bug in InstrumentViewManager

### ISSUE THREAD

**Issue by gui-co** [mantidproject/mantid#31718]
The instrument view manager:
https://github.com/mantidproject/mantid/blob/b23c947d84794dd60c9d5d2c244502b4437aa6d2/qt/python/mantidqt/widgets/instrumentview/presenter.py#L94

maintains an internal dictionary of instrument viewer instances. This dictionary is not properly cleared and old references of closed instrument viewers are kept. You can observe this behavior with this script:

```python
from mantid.simpleapi import *
from mantidqt.widgets.instrumentview.presenter import InstrumentViewManager
from mantidqt.widgets.instrumentview.api import get_instrumentview

ws = CreateSampleWorkspace()
# open and close instrument viewer
i = get_instrumentview(ws)
i.show_view()
i.container.close() # or use the close button

# print the manager internal dictionary
print(InstrumentViewManager.view_dict)
```

I did not find any call to this manager in mantid code but maybe it is used in user's script.
This has been introduced with #29904

**Comment by thomashampson** [comment]
This sample script now causes Mantid to hang. If you try it in on a Windows debug build, the following message is displayed:
![image](https://user-images.githubusercontent.com/8058899/145841585-dd151516-47d7-4c85-9904-81e952f20821.png)

**Comment by ckendrick** [comment]
This sample script should now no longer cause a hang after the fix in #33349. The instrument view was being returned before it was fully initialized by the background thread.

@thomashampson We can probably close this issue now unless this problem is still happening

**Comment by thomashampson** [comment]
> This sample script should now no longer cause a hang after the fix in #33349. The instrument view was being returned before it was fully initialized by the background thread.
> 
> @thomashampson We can probably close this issue now unless this problem is still happening

@ckendrick I'll double check my Windows build on Monday morning 👍

**Comment by martyngigg** [comment]
Fixed by #33015
